tests/dbus/pool/test_udev.py

```diff
@@ -0,0 +1,308 @@
+
+        # Start & Stop the service
+        self._stop_service()
+        self._start_service()
+
+        # We should have all the devices, so pool should exist after toggle
+        self.assertEqual(len(UdevAdd._get_pools()), number_of_pools)
+
+        self._stop_service()
+
+        # Unplug all the devices
+        for device_tokens in pool_data.values():
+            for d in device_tokens:
+                self._lb_mgr.unplug(d)
+
+        self._start_service()
+
+        self.assertEqual(len(UdevAdd._get_pools()), 0)
+
+        # Systematically add a device to each pool, checking that the pool
+        # isn't assembled until complete
+        activation_sequence = []
+
+        pool_names = pool_data.keys()
+
+        for i in range(0, dev_count_pool):
+            for p in pool_names:
+                activation_sequence.append(pool_data[p][i])
+
+        for device_token in activation_sequence[:-number_of_pools]:
+            self._lb_mgr.hotplug(device_token)
+
+            if some_existing:
+                self._stop_service()
+                self._start_service()
+            else:
+                self._settle()
+            result = UdevAdd._get_pools()
+            self.assertEqual(len(result), 0)
+
+        for device_token in activation_sequence[-number_of_pools:]:
+            self._lb_mgr.hotplug(device_token)
+
+        self._settle()
+        self.assertEqual(len(UdevAdd._get_pools()), number_of_pools)
+
+        for pn in pool_names:
+            self.assertEqual(len(self._get_pools(pn)), 1)
+
+        # After this test we need to clean-up in case we are running again
+        # from same test fixture
+        self._stop_service()
+        self._lb_mgr.destroy_devices()
+
+    def test_combinations(self):
+        """
+        Test for 0..2 pools with 1..3 devices in each pool.
+        :return:
+        """
+        for pools_num in range(0, 3):
+            for device_num in range(1, 4):
+                print("Testing %d %d" % (pools_num, device_num))
+                self._test_driver(pools_num, device_num)
+
+    def test_existing(self):
+        """
+        While we are adding devices back we will stop start the daemon to ensure
+        it can start with one or more devices present and complete when the
+        other devices come in later.
+        :return: None
+        """
+        self._test_driver(2, 4, True)
+
+    def _single_pool(self, num_devices, num_hotplugs):
+        """
+        Creates a single pool with specified number of devices.
+        :param num_devices: Number of devices to use for pool
+        :param num_hotplugs: Number of extra udev "add" event per devices
+        :return: None
+        """
+        self._start_service()
+        result = UdevAdd._get_pools()
+        self.assertEqual(len(result), 0)
+
+        device_tokens = []
+
+        for _ in range(0, num_devices):
+            device_tokens.append(self._lb_mgr.create_device(2048))
```

Comment:
With justbytes:

```
Range(2048, MiB)
```

Note that it is pretty easy to tell what this stands for. Enhances readability.